presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java

```diff
@@ -0,0 +1,222 @@
+import io.prestosql.plugin.base.CatalogName;
+import io.prestosql.plugin.hive.HdfsConfig;
+import io.prestosql.plugin.hive.HdfsConfigurationInitializer;
+import io.prestosql.plugin.hive.HdfsEnvironment;
+import io.prestosql.plugin.hive.HdfsEnvironment.HdfsContext;
+import io.prestosql.plugin.hive.HiveHdfsConfiguration;
+import io.prestosql.plugin.hive.authentication.NoHdfsAuthentication;
+import io.prestosql.spi.security.ConnectorIdentity;
+import io.prestosql.testing.TestingNodeManager;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.FilterFileSystem;
+import org.apache.hadoop.fs.LocalFileSystem;
+import org.apache.hadoop.fs.Path;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.reflect.Field;
+import java.net.URI;
+import java.time.Duration;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static com.qubole.rubix.spi.CacheConfig.setPrestoClusterManager;
+import static io.airlift.testing.Assertions.assertGreaterThan;
+import static io.airlift.testing.Assertions.assertInstanceOf;
+import static io.prestosql.client.NodeVersion.UNKNOWN;
+import static io.prestosql.plugin.hive.DynamicConfigurationProvider.setCacheKey;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.nio.file.Files.createDirectories;
+import static java.nio.file.Files.createTempDirectory;
+import static org.apache.parquet.Strings.join;
+import static org.testng.Assert.assertEquals;
+
+@Test(singleThreaded = true)
+public class TestRubixCaching
+{
+    private java.nio.file.Path tempDirectory;
+    private FileSystem nonCachingFileSystem;
+    private FileSystem cachingFileSystem;
+
+    @BeforeClass
+    public void setup()
+            throws IOException
+    {
+        tempDirectory = createTempDirectory(getClass().getSimpleName());
+
+        HdfsConfig config = new HdfsConfig();
+        HdfsConfigurationInitializer configurationInitializer = new HdfsConfigurationInitializer(config);
+        HiveHdfsConfiguration configuration = new HiveHdfsConfiguration(configurationInitializer, ImmutableSet.of());
+        HdfsEnvironment environment = new HdfsEnvironment(configuration, config, new NoHdfsAuthentication());
+        ConnectorIdentity identity = ConnectorIdentity.ofUser("user");
+        HdfsContext context = new HdfsContext(identity);
+        Path path = getPath(".");
```

Comment:
Please use some temp directory for that.

Edit - oh - it actually uses tempDirectory. Plese name method `storagePath()`. And pass `/` to it instead `.`